public-source-ip_test.go

```diff
@@ -35,13 +35,13 @@ func TestIsNotRoutedWithAddressXFFHeaders(t *testing.T) {
 		realip string
 	}{
 		{"10.1.2.3, 206.27.34.1, 192.168.2.200", "206.27.34.1"},
-		{"172.16.255.255, 10.4.0.0.1, 192.1.4.1, 202.0.0.4, 10.1.1.1", "202.0.0.4"},
+		{"172.16.255.255, 10.4.0.0.1, 202.0.0.4, 192.1.4.1, 10.1.1.1", "202.0.0.4"},
```

Comment:
don't change the unit test, if the code doesn't work with this test, that means it will not work correctly in production

Reply:
Should we introduce breaking changes then? Since the behavior definitely changed.
Both 192.1.4.1  and 202.0.0.4 resolves to public IP. So, we need to make sure what is the expected behavior, does it return the closest public IP to the service or closest to the client? Quoting from Mozilla doc

> the right-most IP address is the IP address of the most recent proxy and the left-most IP address is the IP address of the originating client

If it's expected that this return the public IP closest to service, then we cannot change this function to check from left to right.

Reply:
I believe you only have to change the `realip` value to the leftmost/first public IP encountered from client to server

Reply:
@bahrunnur I can do it like what you suggested, but still, I need to update the test.